numpy/ma/core.py

```diff
@@ -245,39 +275,29 @@ def default_fill_value(obj):
     (1e+20+0j)
 
     """
-    if hasattr(obj, 'dtype'):
-        defval = _check_fill_value(None, obj.dtype)
-    elif isinstance(obj, np.dtype):
-        if obj.subdtype:
-            defval = default_filler.get(obj.subdtype[0].kind, '?')
-        elif obj.kind in 'Mm':
-            defval = default_filler.get(obj.str[1:], '?')
+    def _scalar_fill_value(dtype):
+        if dtype.kind in 'Mm':
+            return default_filler.get(dtype.str[1:], '?')
```

Comment:
Yeah, this strikes me as a bad idea as well. It seems like we should just be able to look up the dtype directly in the dictionary anyway.

Either way, I'd like to leave that for a later PR, as like you say, it's a pre-existing line